### DISCUSSION TITLE
Version 4 ?

### DISCUSSION THREAD

**Discussion by cubecleveland** [JedWatson/react-select#4629] [General]
Hello guys,
Where can i find docs about the upgrade form V3 to V4...... 
I am using react-select 3.2.1 and would like to know the changes

**Comment by ebonow** [comment]
Release notes:
https://github.com/JedWatson/react-select/releases/tag/react-select%404.0.0

Also of note, I was attempting to use an issue thread to keep updated release information. It was decided to unpin it but it may be helpful in better describing the changes.

Further information:
https://github.com/JedWatson/react-select/issues/4293#issuecomment-776360081

TLDR; Most of the work was done to make it compatible with React 17 by removing the deprecated lifecycle methods. Breaking changes include onChange returning `[]` instead of `null` for isMulti selects, and cacheKey needing to be added when using NonceProvider.

**Comment by cubecleveland** [comment]
I was wondering about that pesky null..... my code is filled with ?: to
catch :)))))

Did u ever catch the debounce method  when searching in async?
There was bug that there is no debounce on the input when u type in the
search box....

Was that fixed?

P.S

Superb work



CUBE Workspace
Corporate: 6240 Som Center Rd #130
Solon, OH 44139
T: 216.245.9629

**Reply by ebonow** [reply]
@cubecleveland that was considered more as a "feature" than a "bug." 

Fortunately, after some conversation with Jed, it was reasoned and agreed upon by all of us that AsyncSelect is just a wrapper to simplify typeahead look ups from an api and that while we are all trying to be conscious of "prop fatigue", it's very difficult to imagine a use-case where it wouldn't be best-practice to debounce whatever method would be used by loadOptions. 

Therefore, we have this PR https://github.com/JedWatson/react-select/pull/3254 at the top of items to look at once 5.0 gets into and past beta. Feel free to look at that regarding the context of conversation and a suggested workaround". Regarding that timeline, it will likely be this Thursday that the first 5.0 beta is released with us reviewing other potentially breaking changes so that we can integrate as a part of the next major.

**Comment by cubecleveland** [comment]
Great. Keep up the good work.





CUBE Workspace
Corporate: 6240 Som Center Rd #130
Solon, OH 44139
T: 216.245.9629
